server/modules/taxonomy/repository/taxonomyRepository.js

```diff
@@ -34,19 +34,22 @@ export const insertTaxonomy = async (surveyId, taxonomy, client = db) =>
   )
 
 const _insertOrUpdateVernacularNames = (surveyId, taxonUuid, vernacularNames, client = db) =>
-  Object.entries(vernacularNames).map(([lang, vernacularName]) =>
-    client.none(
-      `INSERT INTO 
+  R.pipe(R.values, R.flatten, R.map(vernacularName =>
```

Comment:
why do you need flatten here? can you add a comment on the structure of the vernacular names?

Reply:
ok

Reply:
do you really need to flatten here?

Reply:
vernacularNames is an object indexed by language code. R.values(vernacularNames) gives an array of arrays.